tests/libstorage/datavolume.go

```diff
@@ -224,3 +226,22 @@ func GoldenImageRBAC(namespace string) (*rbacv1.Role, *rbacv1.RoleBinding) {
 	}
 	return role, roleBinding
 }
+
+func RenderVMIWithDataVolume(dvName, ns string, opts ...libvmi.Option) *v1.VirtualMachineInstance {
```

Comment:
This function is a bit odd because creating a VMI in libstorage is funny. Maybe you could refactor it into `VMIDefaults`, and the caller would do:

```
libvmi.New(libstorage.VMIDefaults(name, namespace), opts...)
```

Reply:
I was torn between adding this function here or in `libvmifact` but ended up adding it here since this is a datavolume-specific function for storage tests (in a similar way we have `RenderPodWithPVC` in `pvc.go`).